Re-do "typing for remote_cache"

aorenste opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133401

This is a redo of #133299 which was backed out because of a lint error.

Backout of backout + lint fix

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

aorenste committed:
Re-do "typing for remote_cache"

This is a redo of #133299 which was backed out because of a lint error.

Backout of backout + lint fix

[ghstack-poisoned]

oulgen commented:
You dont need a new PR for reverted PRs, you can upload to same one again and land it

aorenste commented:
> You dont need a new PR for reverted PRs, you can upload to same one again and land it

I know you can do that when the PR starts in OSS but does that work when the PR starts life in phabricator? According to the  instructions on the reverted PR it says to make a new PR.

aorenste committed:
Update on "Re-do "typing for remote_cache""


This is a redo of #133299 which was backed out because of a lint error.

Backout of backout + lint fix

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

aorenste commented:
Abandoning. Realized that I have internal-only changes so this has to be done from phabricator...
